bucket.go

```diff
@@ -299,7 +299,16 @@ func (bucket *Bucket) initializeSchema(bucketName string) (err error) {
 // If the bucket has been closed, it returns a special `closedDB` value that will return
 // ErrBucketClosed from any call.
 func (bucket *Bucket) db() queryable {
-	if db := bucket._db; db != nil {
+	bucket.mutex.Lock()
+	defer bucket.mutex.Unlock()
+	return bucket._db()
+}
+
+// Returns the database handle as a `queryable` interface value.
```

Comment:
Would be good for the comment to point out that this is the same as the `db()` method except for not acquiring the Bucket mutex.